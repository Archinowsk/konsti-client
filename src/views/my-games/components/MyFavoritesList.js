import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { Link } from 'react-router-dom';
import moment from 'moment';

const MyFavoritesList = props => {
  const { favoritedGames, t } = props;

  // Sort games by time and name
  const sortedGames = favoritedGames.sort((a, b) => {
    const keyA = moment(a.date) + a.title;
    const keyB = moment(b.date) + b.title;
    if (keyA < keyB) return -1;
    if (keyA > keyB) return 1;
    return 0;
  });

  const GamesList = sortedGames.map(game => {
    const formattedDate = moment.utc(game.date).format('DD.M.YYYY HH:mm');

    return (
      <li key={game.id}>
        <Link to={`/games/${game.id}`}>
          {formattedDate}: {game.title}
        </Link>
      </li>
    );
  });

  return (
    <div>
      <p>
        {t('favoritedGames')}
      </p>
      <ul>
        {favoritedGames.length === 0 &&
          <span>
            {t('noFavoritedGames')}
          </span>}
        {GamesList}
      </ul>
    </div>
  );
};

MyFavoritesList.propTypes = {
  t: PropTypes.func.isRequired,
  favoritedGames: PropTypes.array.isRequired,
};

export default translate()(MyFavoritesList);
