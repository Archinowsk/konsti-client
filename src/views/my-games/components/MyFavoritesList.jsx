import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { Link } from 'react-router-dom';

const MyFavoritesList = props => {
  const { favoritedGames, t } = props;

  const GamesList = favoritedGames.map(game =>
    <li key={game.id}>
      <Link to={`/games/${game.id}`}>
        {game.title}
      </Link>
    </li>
  );

  return (
    <div>
      <p>
        {t('favoritedGames')}
      </p>
      <ul>
        {GamesList}
      </ul>
    </div>
  );
};

MyFavoritesList.propTypes = {
  t: PropTypes.func.isRequired,
  favoritedGames: PropTypes.array.isRequired,
};

export default translate()(MyFavoritesList);
