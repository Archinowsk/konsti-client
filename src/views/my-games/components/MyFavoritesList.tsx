import React, { FunctionComponent, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';
import { getStartTimes } from 'utils/getStartTimes';
import { GamesByStartTimes } from './GamesByStartTimes';
import { Game } from 'typings/game.typings';

export interface Props {
  favoritedGames: readonly Game[];
}

export const MyFavoritesList: FunctionComponent<Props> = (
  props: Props
): ReactElement<'div'> => {
  const { favoritedGames } = props;
  const { t } = useTranslation();

  const sortedGames: readonly Game[] = _.sortBy(favoritedGames, [
    favoritedGame => favoritedGame.startTime,
    favoritedGame => favoritedGame.title.toLowerCase(),
  ]);

  const startTimes = getStartTimes(favoritedGames);

  return (
    <div className='my-favorites-list'>
      <h3>{t('favoritedGames')}</h3>
      <div className='my-favorites-games'>
        {favoritedGames.length === 0 && <span>{t('noFavoritedGames')}</span>}
        {favoritedGames.length !== 0 && (
          <GamesByStartTimes games={sortedGames} startTimes={startTimes} />
        )}
      </div>
    </div>
  );
};
