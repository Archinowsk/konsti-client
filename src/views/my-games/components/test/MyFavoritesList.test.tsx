import React from 'react';
import { shallow } from 'enzyme';
import { MyFavoritesList, Props } from '../MyFavoritesList';

const favoritedGames = [];

describe('MyFavoritesList', () => {
  it('should render correctly', () => {
    const props: Props = { favoritedGames };
    const component = shallow(<MyFavoritesList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
