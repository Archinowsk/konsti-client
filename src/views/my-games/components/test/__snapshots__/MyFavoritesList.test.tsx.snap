// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavoritesList should render correctly 1`] = `
<div
  className="my-favorites-list"
>
  <h3>
    favoritedGames
  </h3>
  <div
    className="my-favorites-games"
  >
    <span>
      noFavoritedGames
    </span>
  </div>
</div>
`;
